### ISSUE TITLE
[Progress] - Option for progress inline label centered on the whole bar

### ISSUE THREAD

**Issue by dabroz** [Semantic-Org/Semantic-UI#3819]
Could we have an option to always display progress inline label centered on the whole bar, and not only the completed part? Currently this causes small progress bars not to display progress label correctly. Compare ui-progress with a custom solution:

![Compare](http://i.imgur.com/sYXWXpK.png)

Having a separate label instead of the inline one makes the progress bar take more vertical space.

**Comment by jlukic** [comment]
The only issue with that is that it would need to correctly invert colors as it was passed by the progress.

**Comment by dabroz** [comment]
Agreed! But even a simple black label would be visible on all default bars (regular, error etc) if the color switching in between would happen to be too cumbersome to implement.

**Comment by dabroz** [comment]
I've found another issue. When using small progress bars, the width does not reflect progress for low values. It looks like a space was left for the inline label. However, the inline label itself is not being displayed.

![Progress](http://i.imgur.com/hLJx3TT.png)

Code used:

    <div class='ui small progress' data-percent='0'>
      <div class='bar'>
        <div class='progress'></div>
      </div>
      <div class='label'>0%</div>
    </div>
    <div class='ui small progress' data-percent='8'>
      <div class='bar'>
        <div class='progress'></div>
      </div>
      <div class='label'>8%</div>
    </div>
    <div class='ui small progress' data-percent='56'>
      <div class='bar'>
        <div class='progress'></div>
      </div>
      <div class='label'>56%</div>
    </div>

**Comment by jlukic** [comment]
As a stop gap you can adjust the bars `min-width` to prevent issues with small percentages in small bars

**Comment by bniehuser** [comment]
the latter part of this issue is still present in all implementations.  also, progress bar overrun is inconsistent; in some presentations it caps at bar length and others it exceeds container width.
when overrun is possible (usually when progress is deeply nested in other presentation elements) it also renders progress label outside bar at low values (<10%, or whatever makes progress label width larger than available bar width).  i can use the 'min-width' fix (and presumably an 'overflow: hidden' fix) to clean these up in my implementation, however i think the original suggestion of on-bar labeling (both progress and/or label itself) is worth looking into for applications which require several bars in condensed space, especially where label/progress indicator visiblity are needed for all possible (including out of range) progress values.  consider it another 2 cents; overall semantic ui is awesome but this module could be improved.

**Comment by lubber-de** [comment]
The Fomantic-UI fix in https://github.com/fomantic/Fomantic-UI/pull/412 covers at least the 0% width display now
